src/Common/AzurePSCmdlet.cs

```diff
@@ -880,6 +883,13 @@ protected void LogQosEvent()
             _qosEvent.ParameterSetName = this.ParameterSetName;
             _qosEvent.FinishQosEvent();
 
+            if (!AzureSession.Instance.TryGetComponent(AuthenticationTelemetry.Name, out AuthenticationTelemetry authenticationTelemetry))
+            {
+                throw new NullReferenceException(Resources.AuthenticationTelemetryNotRegistered);
```

Comment:
1. `InvalidOperationException` may be a better choice
2. If cx disables telemetry, will this exception be thrown?

Reply:
I have checked existing codes. When `IsDataCollectionAllowed()` returns false, the data collection related modules are still registered but the telemetry data is not sent. 
Specifically, in the function you refers to (`LogQosEvent`), all the codes are no need executing when `IsDataCollectionAllowed()` returns false. But I am not sure whether there is a side effect to skip the operations in this function when data collection is not allowed. So I keep the existing null reference check to respect the current implementation.